src/scripts/main.js

```diff
@@ -1,3 +1,317 @@
 'use strict';
 
-// write code here
+// 1. Implement table sorting by clicking on the title (in two directions).
+const table = document.querySelector('table');
+
+function getTableRows() {
+  return [...table.querySelectorAll('tbody tr')];
+}
+
+function mapTableData(rows) {
+  return rows.map((row) => {
+    return [...row.children].map((cell) => cell.innerText);
+  });
+}
+
+function getTableData() {
+  const rows = getTableRows();
+
+  return mapTableData(rows);
+}
+
+let sortDirection = true;
+
+function sortTable(columnIndex) {
+  const sortedData = sortTableData(columnIndex, sortDirection);
+
+  renderSortedTable(sortedData);
+  toggleSortDirection();
+}
+
+function sortTableData(columnIndex, direction) {
+  const data = getTableData();
+
+  return data.sort((rowA, rowB) => {
+    let cellA = rowA[columnIndex];
+    let cellB = rowB[columnIndex];
+
+    cellA = parseCellData(cellA, columnIndex);
+    cellB = parseCellData(cellB, columnIndex);
+
+    if (direction) {
+      return cellA > cellB ? 1 : -1;
+    }
+
+    return cellA > cellB ? -1 : 1;
+  });
+}
+
+function parseCellData(cell, columnIndex) {
+  if (columnIndex === 3) {
+    return parseInt(cell);
+  }
+
+  if (columnIndex === 4) {
+    return parseFloat(cell.slice(1).replace(/,/g, ''));
+  }
+
+  return cell;
+}
+
+function renderSortedTable(sortedData) {
+  const tableBody = table.querySelector('tbody');
+
+  tableBody.innerHTML = '';
+
+  sortedData.forEach((rowData) => {
+    const row = document.createElement('tr');
+
+    rowData.forEach((cellData) => {
+      const cell = document.createElement('td');
+
+      cell.textContent = cellData;
+      row.appendChild(cell);
+    });
+    tableBody.appendChild(row);
+  });
+}
+
+function toggleSortDirection() {
+  sortDirection = !sortDirection;
+}
+
+function attachSortingListeners() {
+  table.querySelectorAll('th').forEach((header, index) => {
+    header.addEventListener('click', () => sortTable(index));
+  });
+}
+
+attachSortingListeners();
+
+// 2. When user clicks on a row, it should become selected.
+let activeRow = null;
+const tbody = table.querySelector('tbody');
+
+tbody.addEventListener('click', (e) => {
+  const row = e.target.closest('tr');
+
+  if (!row) {
+    return;
+  }
+
+  if (activeRow) {
+    activeRow.classList.remove('active');
+  }
+
+  row.classList.add('active');
+  activeRow = row;
+});
```

Comment:
Code quality issue: You are directly manipulating the DOM to set the active row. Consider using a more declarative approach, for example by storing the active row in a state variable and using this to update the DOM.